src/Resources/js/modules/participation-update-handler.ts

```diff
@@ -81,12 +82,13 @@ export class ParticipationUpdateHandler {
         this.updateCheckBoxWrapper($checkbox);
     }
 
-    public static toggleAction($checkbox: JQuery, actionText: string, url: string, participantsCount: number, bookedDishIDs: string[]) {
+    // public static toggleAction($checkbox: JQuery, actionText: string, url: string, participantsCount: number, bookedDishIDs: string[]) {
+    public static toggleAction($checkbox: JQuery, res: ToggleResponse) {
```

Comment:
I changed it because the argument list is getting too big now. What's the alternative?

Reply:
The problem is, you can't initialize an interface (new ToggleResponse()), so toggleAction method could only be used by an ajax callback function (where you expect the result to be ToggleResponse). On "fix" would be, if you change ToggleResponse from interface to class but it's not so nice if you would like to toggle the action in the future (for any reason) from code, you need to create a (Toggle)Response object. Then the dev will ask "Why a ToggleResponse object", I didn't make any request?! 
So the idea of the UpdateHandler was to be independent of ajax (which will be removed in the future somewhen). 
In my opinion, the alternative is a long parameter list in the toggleAction or you split the toggleAction in multiple methods, but then the toggle is not like a "transaction" anymore (The state of checkbox, participant counter, etc. should change at "once"  or at least with on method call because it's all connected).
Is a long parameter list that bad? :-)

Reply:
Okay, I changed it slightly. Instead of ToggleResponse object, we use ToggleData object as parameter now.

Long parameter list is considered as a code smell. :)